example/optimus/coordinator/coordinator.go

```diff
@@ -0,0 +1,96 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/edegens/ptype/cluster"
+	prime "github.com/edegens/ptype/example/optimus"
+)
+
+var worker *cluster.Client
+
+func main() {
+	cfg, err := cluster.ConfigFromFile(os.Getenv("CONFIG"))
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	c, err := cluster.Join(context.Background(), cfg)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	services, err := c.Registry.Services(context.Background())
+	if err != nil {
+		log.Fatal(err)
+	}
+	fmt.Printf("coordinator: services %v\n", services)
+
+	// let the worker spin up after etcd
+	time.Sleep(500 * time.Millisecond)
+	worker, err = c.NewClient("prime_worker")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	http.HandleFunc("/test", test)
+	log.Fatal(http.ListenAndServe(":8082", nil))
+}
+
+func test(w http.ResponseWriter, r *http.Request) {
```

Comment:
I would make a struct and make these functions methods of the struct. This way we can have the client as an instance variable instead of a global variable.

Reply:
you are right but does it mater in such a trivial case?

Reply:
I'd argue that it's a better showcase for a library if the examples follow good practices.